Timer.DAL/Timer.DAL/UnitOfWork/UnitOfWork.cs

```diff
@@ -9,47 +9,22 @@ namespace Timer.DAL.Timer.DAL.UnitOfWork
 {
     public class UnitOfWork : IUnitOfWork
     {
-        private ApplicationUserManager userManager;
-        public ApplicationUserManager UserManager
-        {
-            get
-            {
-                if (userManager == null)
-                {
-                    userManager = new ApplicationUserManager(new UserRepository(context));
-                }
-                return userManager;
-            }
-            set
-            {
-                userManager = value;
-            }
-        }
+        public IApplicationUserManager<User,int> UserManager {get;set; }
 
-        private IRepository<Role> roles;
-        public IRepository<Role> Roles {
-            get
-            {
-                if(roles == null)
-                {
-                    roles = new Repository<Role>(context);
-                }
-                return roles;
-            }
-            set
-            {
-                roles = value;
-            }
-        }
+        public IRepository<Role> Roles {get;set;}
 
         private TimerContext context;
+
         private bool disposed;
 
-        public UnitOfWork(TimerContext context)
+        public UnitOfWork(TimerContext context, IApplicationUserManager<User,int> applicationUserManager,IRepository<Role> role)
         {
             this.context = context;
             this.disposed = false;
+            this.UserManager = applicationUserManager;
+            this.Roles = role;
```

Comment:
this is still can be made lazy using method I mentioned in https://github.com/ss-ita/gtd-timer-server/issues/65

1. make Lazy backing field for property
2. actual property should be of interface type and have setter like this:

(taking roles as example)
set { roles = new Lazy<IRepository<Role>>(() => value)
3. constructor can stay as is